ranzen/misc.py

```diff
@@ -147,3 +160,88 @@ def _generate_next_value_(name: str, start: int, count: int, last_values: list[A
             Return the lower-cased version of the member name.
             """
             return name.lower()
+
+
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+
+class AddDict(Dict[_KT, _VT], Addable):
+    """
+    Extension of the built-in dictionary class that supports use of the ``__add__`` operator for
+    merging its values with those of other dictionaries. Note that, for the sake of simplicity,
+    addition requires both dictionaries to be of the same generic type. Values that do not have an
+    ``__add__`` operator defined, either in general or with respect to the other value, will be
+    merged into a list
+
+    :example:
+
+    .. code-block:: python
+        # Simple case of addition of integers.
+        d1 = AddDict({"foo": 1, "bar": 2})
+        d2 = {"foo": 3, "bar": 4}
+        d1 + d2 # {'foo': 4, 'bar': 6}
+
+        # ... Concatenation of lists
+        d3 = AddDict({"foo": [1], "bar": [2]})
+        d4 = {"foo": [3, 4], "bar": 4}
+        d3 + d4 # {'foo': [1, 3, 4], 'bar': [2, 4]}
+    """
+
+    @overload
+    def __add__(self: Self, other: int) -> Self:
+        ...
+
+    @overload
+    def __add__(self: Self, other: dict[_KT, _VT]) -> AddDict[_KT, _VT | list[_VT]]:
+        ...
+
+    def __add__(
+        self: Self,
+        other: int | dict[_KT, _VT],
+    ) -> Self | AddDict[_KT, _VT | list[_VT]]:
+        # Allow ``other`` to be an integer, but specifying the identity function, for compatibility
+        # with # the 'no-default' version of``sum``.
+        if isinstance(other, int):
+            return self
+        copy: AddDict[_KT, Union[_VT, list[_VT]]] = AddDict()
+        copy.update(gcopy(self, deep=False))
+        for key_o, value_o in other.items():
+            if key_o in self:
+                value_s = self[key_o]
+                if isinstance(value_s, Addable) and isinstance(value_o, Addable):
+                    try:
+                        copy[key_o] = value_s + value_o
+                    except TypeError:
+                        copy[key_o] = [value_s, value_o]
+                else:
+                    copy[key_o] = [value_s, value_o]
+            else:
+                copy[key_o] = value_o
+        return copy
+
+    @overload
+    def __radd__(self: Self, other: int) -> Self:
+        ...
+
+    @overload
+    def __radd__(self: Self, other: dict[_KT, _VT]) -> AddDict[_KT, _VT | list[_VT]]:
+        ...
+
+    def __radd__(self: Self, other: int | dict[_KT, _VT]) -> Self | AddDict[_KT, _VT | list[_VT]]:
+        return self + other
+
+
+A = TypeVar("A", bound=Addable)
+
+
+def reduce_add(__iterable: Iterable[A]) -> A:
```

Comment:
Why the double underscore? If you wanted to make the function work with positional arguments only, you can use `/`

Reply:
```
@overload
def sum(__iterable: Iterable[_SupportsSumNoDefaultT]) -> _SupportsSumNoDefaultT | Literal[0]: ...
```

Reply:
Huh, why is that so weird?